config/config-api/src/main/java/com/thoughtworks/go/config/AgentConfig.java

```diff
@@ -181,26 +181,67 @@ public boolean isFromLocalHost() {
         return cachedIsFromLocalHost.booleanValue();
     }
 
-    public String getIpAddress() {
-        return this.ipAddress;
+    public boolean isElastic() {
+        return isNotBlank(elasticAgentId) && isNotBlank(elasticPluginId);
```

Comment:
It would be nice to have polymorphic objects like `ElasticAgent extends Agent`. I know it might cascade into a lot of changes everywhere else, so feel free to ignore this for now.